### ISSUE TITLE
[a11y][Modal] Missing aria-modal attribute

### ISSUE THREAD

**Issue by ddayguerrero** [Shopify/polaris#2393]
## Issue summary

[Modal](https://polaris.shopify.com/components/overlays/modal) does not indicate its presence; preventing users from understanding the purpose of the current content they're in.

## Expected behavior

Add missing `aria-modal` to the `<Modal/>` Polaris component. 

According to [WAI-ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/#aria-modal):

> - Assistive technologies SHOULD navigate to the element (modal) unless focus has explicitly been set elsewhere
> - Assistive technologies MAY limit navigation to the modal element's contents

## Actual behavior

- `<Modal/> is missing the `aria-modal="true"` attribute

## Steps to reproduce the problem

1. Render a example Modal component
2. Inspect markup with missing `aria-modal` attribute

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 4.7.2
- Browser: Safari, Chrome
- Device: MBP 15
- Operating System: Mac OS

**Comment by ddayguerrero** [comment]
While Polaris documentation mentions the following:

> Modals use ARIA role=”dialog” to convey to screen reader users that they work like native dialog windows.

I think that the required `aria-modal` inherits into the `dialog` role

**Comment by dpersing** [comment]
Hi @ddayguerrero! Thanks for the issue. The `aria-modal` property is quite new (at least in spec and assistive tech support terms), and at the time the audit you point to was done, support was at zero. I lumped it under "enhancements currently available in the 1.1 spec for modals, but support is either not present or buggy at present".

According to the current ARIA practices documentation, the `aria-modal` attribute is meant to replace `aria-hidden` to hide content "underneath" from the user. I do think that adding the attribute won't hurt (since it adds to the information provided by the `role`), but I do think it's lower priority than the focus management and other issues in the modal issue you linked to, and is encompassed in pointing folks to the spec in the recommendations.

There is currently work on improving the modal underway, and more closely aligning with the 1.1 spec is definitely a goal. 👍

**Comment by ddayguerrero** [comment]
@dpersing, thanks for the feedback! This issue was brought up as a result of an audit we had. While the fix for this issue is a quick win, I think @svinkle might have better context on the impact of such change. I don't mind holding off on this change in the meantime.

That said, where can we follow the current work in progress for modal (aside from existing issues)?

**Comment by dpersing** [comment]
@ddayguerrero Unfortunately the modal issues and their attached PRs are the only thing to track at the moment. The work has been ongoing as part of a push to fix accessibility issues in the backlog. That work is going to be paused in general to help with some larger team projects, but issues and PRs will all remain open as long as they need to be!

**Comment by alex-page** [comment]
We will explore this in the future with thorough accessibility testing. Closing for now as I am unsure that adding `aria-modal` is going to enhance the experience for users.

**Comment by alex-page** [comment]
@svinkle happy to reopen this if there is any context you can add.

**Comment by svinkle** [comment]
@alex-page Including `aria-model="true"` on the dialog container alerts AT that a modal is present. For screen readers specifically, it helps to keep virtual cursors within the modal context. Like how we need to manually prevent <kbd>Tab</kbd> key press from escaping, this does the same but "automatically" for virtual cursors. Definitely want to add this.

**Comment by kyledurand** [comment]
Looks like this was already added @svinkle https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/Modal/components/Dialog/Dialog.tsx#L72

Let me know if it's still not working as expected
